hibernate-core/src/main/java/org/hibernate/dialect/FirebirdDialect.java

```diff
@@ -266,27 +266,6 @@ public void initializeFunctionRegistry(QueryEngine queryEngine) {
 				"((?1)*180e0/PI())",
 				StandardBasicTypes.DOUBLE
 		);
-		functionRegistry.namedDescriptorBuilder( "rdb$get_context" )
-				.setExactArgumentCount( 2 )
-				.setInvariantType( StandardBasicTypes.STRING )
-				.setArgumentListSignature( "(namespace, varname)" );
-		functionRegistry.namedDescriptorBuilder( "rdb$set_context" )
-				.setExactArgumentCount( 3 )
-				.setInvariantType( StandardBasicTypes.INTEGER )
-				.setArgumentListSignature( "(namespace, varname, value)" );
-		functionRegistry.namedDescriptorBuilder( "char_to_uuid" )
-				.setExactArgumentCount( 1 )
-				.setInvariantType( StandardBasicTypes.UUID_BINARY );
-		functionRegistry.namedDescriptorBuilder( "gen_uuid" )
-				.setExactArgumentCount( 0 )
-				.setInvariantType( StandardBasicTypes.UUID_BINARY );
-		functionRegistry.namedDescriptorBuilder( "uuid_to_char" )
-				.setExactArgumentCount( 1 )
-				.setInvariantType( StandardBasicTypes.UUID_CHAR );
-		functionRegistry.namedDescriptorBuilder( "gen_id" )
-				.setExactArgumentCount( 2 )
-				.setInvariantType( StandardBasicTypes.BIG_INTEGER )
-				.setArgumentListSignature( "(sequence, step)" );
```

Comment:
Same for these functions. What do you think @gavinking?

Reply:
These functions aren't generally useful from *queries*.

Reply:
But again, if someone wants to go through *all* the `Dialect`s which support a UUID-like type, and define a portable `uuid_to_char()` function, that would be OK.

On the other hand, it would be much better to just make sure that you can do it using the existing `cast()` function, i..e

    cast(uuid to String)
    cast(char to UUID)

Reply:
I can add them back in if there is a consensus to do so.

Reply:
Firebird doesn't actually have a specific UUID type (so mapping to `StandardBasicTypes.UUID_BINARY` might have been the wrong thing). Basically the `char_to_uuid`/`uuid_to_char` functions transform between human 'readable' UUID string (usually `char(36)`) and `binary(16)` (with the assumption that is a binary UUID).

In any case, lets leave them out, we can always revisit this if there turns out to be a strong need for this.